modules/standard/DateTime.chpl

```diff
@@ -0,0 +1,1571 @@
+      return DAYS_IN_MONTH(month);
+  }
+
+/* A record representing a date */
+  record date {
+    pragma "no doc"
+    var chpl_year, chpl_month, chpl_day: int;
+
+    /* The year represented by this date value */
+    proc year {
+      return chpl_year;
+    }
+
+    /* The month represented by this date value */
+    proc month {
+      return chpl_month;
+    }
+
+    /* The day represented by this date value */
+    proc day {
+      return chpl_day;
+    }
+
+    /* The minimum representable date */
+    proc type min {
+      return new date(MINYEAR, 1, 1);
+    }
+
+    /* The maximum representable date */
+    proc type max {
+      return new date(MAXYEAR, 12, 31);
+    }
+
+    /* The minimum non-zero difference between two dates */
+    proc type resolution {
+      return new timedelta(days=1);
+    }
+  }
+
+
+  /* constructors/factories for date values */
+
+  /* Construct a new date value from a year, month, and day. All
+     three arguments are required and must be in valid ranges.  The
+     valid ranges are:
+
+     1 <= year <= 9999
+     1 <= month <= 12
+     1 <= day <= the number of days in the given month and year
+  */
+  proc date.date(year, month, day) {
+    if year < MINYEAR-1 || year > MAXYEAR+1 then
+      halt("year is out of the valid range");
+    if month < 1 || month > 12 then
+      halt("month is out of the valid range");
+    if day < 1 || day > daysInMonth(year, month) then
+      halt("day is out of the valid range");
+
+    this.chpl_year = year;
+    this.chpl_month = month;
+    this.chpl_day = day;
+  }
+
+  /* A date object representing the current day */
+  proc type date.today() {
+    const timeSinceEpoch = getTimeOfDay();
+    const td = new timedelta(seconds=timeSinceEpoch(1),
+                             microseconds=timeSinceEpoch(2));
+
+    return unixEpoch.getdate() + td;
+  }
+
+  /* The date that is `timestamp` seconds from the epoch */
+  proc type date.fromtimestamp(timestamp) {
+    const sec = timestamp: int;
+    const us = ((timestamp-sec) * 1000000 + 0.5):int;
+    const td = new timedelta(seconds=sec, microseconds=us);
+    return unixEpoch.getdate() + td;
+  }
+
+  /* The date that is `ord` days from 1-1-0001 */
+  proc type date.fromordinal(ord) {
+    if ord < 0 || ord > 1+date.max.toordinal() then
+      halt("ordinal (", ord, ") out of range");
+    const (y,m,d) = ordToYmd(ord);
+    return new date(y,m,d);
+  }
+
+
+  /* Methods on date values */
+
+  /* Replace the year, month and/or day in a date to create a new date */
+  proc date.replace(year=0, month=0, day=0) {
+    const newYear = if year > 0 then year else this.year;
+    const newMonth = if month > 0 then month else this.month;
+    const newDay = if day > 0 then day else this.day;
+    return new date(newYear, newMonth, newDay);
+  }
+
+  /* Return a filled record matching the C "struct tm" type for the given date */
+  proc date.timetuple() {
+    var timeStruct: tm;
+
+    // no 0 init for extern records?
```

Comment:
I think I may have been mistaken.  I tried a standalone test and it appears to zero the fields.  I think I'll just remove that comment.